tests/TestCase/Validation/ValidationTest.php

```diff
@@ -2813,6 +2817,9 @@ public function testContainNonAlphaNumeric()
         $this->assertFalse(Validation::containsNonAlphaNumeric('abcdef##', 3));
         $this->assertFalse(Validation::containsNonAlphaNumeric('abc##def', 3));
         $this->assertFalse(Validation::containsNonAlphaNumeric('ab#cd#ef', 3));
+
+        //CHECK SCALAR
```

Comment:
But arent you doing the opposite here? Checking a non-scalar (array) value?
Usually one can make a separate test method with the method name making it clear, also optionally adding a doc block note there.